apps/src/code-studio/components/header/SignInCalloutLogic.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import cookies from 'js-cookie';
+import SignInCallout from './SignInCallout';
+
+const HideSignInCallout = 'hide_signin_callout';
+
+// The use of both session storage and cookies is to check for 1 day
+// and 1 session, and display the callout again only once BOTH have passed.
+export default class SignInCalloutLogic extends React.Component {
+  constructor(props) {
+    super(props);
+    this.closeCallout = this.closeCallout.bind(this);
+    this.state = {
+      showCallout: !(
+        cookies.get(HideSignInCallout) === 'true' ||
+        sessionStorage.getItem(HideSignInCallout) === 'true'
+      )
```

Comment:
Just a thought, it may be easier to follow this logic if the state is `hideCallout = cookies.get(HideSignInCallout) === 'true' || sessionStorage.getItem(HideSignInCallout) === 'true'` instead of `showCallout`

Reply:
I like this- will implement now